src/EventSubscriber/RequestSubscriber.php

```diff
@@ -174,7 +171,11 @@ public function onFinishRequest(FinishRequestEvent $event) {
 
     // End the transaction.
     try {
-      $this->phpAgent->stopTransaction($this->routeMatch->getRouteName());
+      $transaction = $this->phpAgent->getTransaction($this->routeMatch->getRouteName());
+      // Capture database time by wrapping spans around the db queries run.
+      $transaction->setSpans($this->constructDatabaseSpans());
```

Comment:
Does this make sense to do? Looks like the transaction would only have spans related to database queries since it overwrites whatever the transaction's spans were before - but it doesn't look like \PhilKra\Events\Transaction supports appending spans.

Reply:
Overwriting spans in the transaction may be a non-issue in the version of elastic-apm-php-agent that supports v2 of the api - the sample at https://github.com/philkra/elastic-apm-php-agent/issues/77 suggests you can add spans as you go.

This kind of thing looks easier in https://github.com/frankkoornstra/elastic-apm-agent except there's no indication in the issue queue that v2 support is being worked on.